contracts/src/GaspToken.sol

```diff
@@ -1,30 +1,112 @@
-// SPDX-License-Identifier: MIT
-pragma solidity ^0.8.12;
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.13;
 
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import {IGaspToken} from "./interfaces/IGaspToken.sol";
 
-contract MintableERC20 is ERC20 {
-    address public owner;
-    constructor(string memory name_, string memory symbol_) ERC20(name_, symbol_) {
-        owner = msg.sender;
-        _mint(msg.sender, 1000000000 * 10 ** decimals());
+contract GaspToken is Context, Ownable, ERC20, IGaspToken {
+    uint256 private constant _TOTAL_SUPPLY = 1_000_000_000 * 10 ** 18;
```

Comment:
The total supply should be 1B, not 10B
See https://www.notion.so/mangata-finance/Non-Tradable-GASP-Token-146e667472b58042a14ac392def43826?pvs=4#146e667472b580bb9327feb2b7265284

Also, we need to state amount of decimals in the contract for the sake of readability and proper handling by thirdparty frontends (maybe it is stated somewhere in the dependencies but I see 18 as a constant here)

Reply:
Please disregard the amount part, I've miscalculated it.